### PR TITLE
Fix clusterViewListener Leak

### PR DESCRIPTION
**Pull request by sancar**
If connection closes while registering listener, we can get
CancelledKeyException from initial push operation. Since we can
not add the necessary DestroyAction it was causing the leak.

Added catch and ignore for CancelledKeyException to avoid the problem.

fixes https://github.com/hazelcast/hazelcast/issues/16482

### CODE REVIEW

**Comment by sancar** [comment]
```
java.nio.channels.CancelledKeyException
	at sun.nio.ch.SelectionKeyImpl.ensureValid(SelectionKeyImpl.java:73)
	at sun.nio.ch.SelectionKeyImpl.interestOps(SelectionKeyImpl.java:77)
	at com.hazelcast.internal.networking.nio.NioPipeline.registerOp(NioPipeline.java:143)
	at com.hazelcast.internal.networking.nio.NioOutboundPipeline.executePipeline(NioOutboundPipeline.java:243)
	at com.hazelcast.internal.networking.nio.NioOutboundPipeline.write(NioOutboundPipeline.java:215)
	at com.hazelcast.internal.networking.nio.NioChannel.write(NioChannel.java:79)
	at com.hazelcast.internal.nio.tcp.TcpIpConnection.write(TcpIpConnection.java:185)
	at com.hazelcast.client.impl.ClusterViewListenerService.registerListener(ClusterViewListenerService.java:112)
	at com.hazelcast.client.impl.protocol.task.AddClusterViewListenerMessageTask.call(AddClusterViewListenerMessageTask.java:45)
	at com.hazelcast.client.impl.protocol.task.AbstractCallableMessageTask.processMessage(AbstractCallableMessageTask.java:35)
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.initializeAndProcessMessage(AbstractMessageTask.java:145)
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.run(AbstractMessageTask.java:108)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64)
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80)
```

This is the stacktrace which helped me find the cause.

File: hazelcast/src/main/java/com/hazelcast/client/impl/ClusterViewListenerService.java [hunk 1]
```diff
@@ -87,7 +89,17 @@ private void sendToListeningEndpoints(ClientMessage clientMessage) {
             ClientMessage message = clientMessage.copyWithNewCorrelationId(correlationId);
             ClientEndpoint clientEndpoint = entry.getKey();
             Connection connection = clientEndpoint.getConnection();
+            write(message, connection);
```
Comment (mdumandag) [inline hazelcast/src/main/java/com/hazelcast/client/impl/ClusterViewListenerService.java]: Why are we also catching the exception here ? I get the reasoning of doing it on the register listener part but not here

Comment (sancar) [reply hazelcast/src/main/java/com/hazelcast/client/impl/ClusterViewListenerService.java]: If we don't catch exception for one endpoint, we can not send events to other healthy endpoints

**Comment by sancar** [comment]
We have unrelated test failures on Membership failures tests
http://jenkins.hazelcast.com/job/Hazelcast-pr-builder/5654/

**Pull request merged**